lib/smart_proxy_remote_execution_ssh/runners/script_runner.rb

```diff
@@ -170,8 +170,10 @@ def initialization_script
       su_method = @user_method.instance_of?(SuUserMethod)
       # pipe the output to tee while capturing the exit code in a file
       <<-SCRIPT.gsub(/^\s+\| /, '')
-      | sh -c "(#{@user_method.cli_command_prefix}#{su_method ? "'#{@remote_script} < /dev/null '" : "#{@remote_script} < /dev/null"}; echo \\$?>#{@exit_code_path}) | /usr/bin/tee #{@output_path}
-      | exit \\$(cat #{@exit_code_path})"
+      | sh <<EOF | /usr/bin/tee #{@output_path}
+      | #{@user_method.cli_command_prefix}#{su_method ? "'#{@remote_script} < /dev/null '" : "#{@remote_script} < /dev/null"}; echo \\$?>#{@exit_code_path};
+      | exit \\$(cat #{@exit_code_path})
+      | EOF
```

Comment:
Maybe just a matter of personal taste, but could you use a squiggly heredoc here instead of a regular heredoc with substitution?